astropy/coordinates/tests/test_frames.py

```diff
@@ -1139,3 +1139,63 @@ class NameChangeFrame(BaseCoordinateFrame):
 
     # Check that the letter "r" has not been replaced more than once in the Frame repr
     assert repr(frame).count("JUSTONCE") == 1
+
+
+def test_galactocentric_defaults():
+    from astropy.coordinates import (Galactocentric,
+                                     galactocentric_frame_defaults,
+                                     BaseCoordinateFrame,
+                                     CartesianDifferential)
+
+    # To restore this at end:
+    init_value = galactocentric_frame_defaults._value
+
+    with galactocentric_frame_defaults.set('pre-v4.0'):
+        galcen_pre40 = Galactocentric()
+
+    with galactocentric_frame_defaults.set('v4.0'):
+        galcen_40 = Galactocentric()
+
+    with galactocentric_frame_defaults.set('latest'):
+        galcen_latest = Galactocentric()
+
+    # parameters that changed
+    assert not u.allclose(galcen_pre40.galcen_distance,
+                          galcen_40.galcen_distance)
+    assert not u.allclose(galcen_pre40.z_sun, galcen_40.z_sun)
+
+    for k in galcen_40.get_frame_attr_names():
+        if isinstance(getattr(galcen_40, k), BaseCoordinateFrame):
+            continue  # skip coordinate comparison...
+
+        elif isinstance(getattr(galcen_40, k), CartesianDifferential):
+            assert u.allclose(getattr(galcen_40, k).d_xyz,
+                              getattr(galcen_latest, k).d_xyz)
+        else:
+            assert getattr(galcen_40, k) == getattr(galcen_latest, k)
+
+    galactocentric_frame_defaults._value = init_value
+
+
+def test_galactocentric_default_warning():
+    from astropy.coordinates import (Galactocentric,
+                                     galactocentric_frame_defaults)
+
+    # TODO: this needs to be audited and removed if we change the default
+    # defaults (yes) for v4.1. This is a hack to get around the fact that the
+    # doctests in galactocentric.py actually change the *class* state:
```

Comment:
Still a bit of a hack, but maybe you can add a ``.. testsetup::`` clause in the docs that resets the science state to the default setting?  Then this wouldn't be necessary and more generally would prevent unintended consequences of the doctests...

Reply:
Good point! I think we want `testcleanup` but yes, see latest commit!